edpop_explorer/readers/kb.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass, field as dataclass_field
+from typing import Optional
+import yaml
+from edpop_explorer.srumarc21reader import SRUReader, APIRecord
+
+
+@dataclass
+class KBRecord(APIRecord):
+    data: dict = dataclass_field(default_factory=dict)
+    identifier: Optional[str] = None
+
+    def get_title(self) -> str:
+        if 'title' in self.data:
+            title = self.data['title']
+            if type(title) == list:
+                # Title contains a list of strings if it consists of multiple
+                # parts
+                return ' : '.join(title)
+            else:
+                return title
+        else:
+            return '(no title defined)'
+
+    def show_record(self) -> str:
+        return_string = yaml.safe_dump(self.data, allow_unicode=True)
+        if self.link:
+            return_string = self.link + '\n' + return_string
+        return return_string
+
+
+class KBReader(SRUReader):
+    sru_url = 'http://jsru.kb.nl/sru'
+    sru_version = '1.2'
+    KB_LINK = 'https://webggc.oclc.org/cbs/DB=2.37/PPN?PPN={}'
+
+    def __init__(self):
+        super().__init__()
+        # Set in init method because dicts are mutable
+        self.additional_params = {
+            'x-collection': 'GGC'
+        }
+
+    def transform_query(self, query: str) -> str:
+        return query
+
+    def _find_ppn(self, data: dict):
+        """Try to find the PPN given the data that comes from the SRU server"""
+        # This seems to work fine; not thoroughly tested.
+        oaipmhidentifier = data.get('OaiPmhIdentifier', None)
```

Comment:
Don't bother changing this here, but it does help readability to use some underscores in long variable names like this, e.g. `oai_pmh_identifier`.